server/eclipse-project/src/main/java/au/org/democracydevelopers/corla/endpoint/GetAssertions.java

```diff
@@ -21,5 +21,258 @@
+     * - For each IRV contest, make a request to the raire-service get-assertions endpoint of the right format type
+     * - Collate all the results into a zip
+     * @param zos an output stream (to become a zip file)
+     * @param riskLimit the risk limit
+     * @param raireUrl the url where the raire-service is running
+     * @param suffix requested file type: "csv" or "json"
+     */
+    public void getAssertions(final ZipOutputStream zos, final BigDecimal riskLimit, String raireUrl, String suffix) throws IOException {
+        final String prefix = "[getAssertions]";
+
+        // Iterate through all IRV Contests, sending a request to the raire-service for each one's assertions and
+        // collating the responses.
+        final List<ContestResult> IRVContestResults = IRVContestCollector.getIRVContestResults();
+        for (ContestResult cr : IRVContestResults) {
+            //  IRVContestResults.forEach(cr -> {
+
+            // Find the winner (there should only be one), candidates and contest name.
+            // TODO At the moment, the winner isn't yet set properly - will be set in the GenerateAssertions Endpoint.
+            // See https://github.com/DemocracyDevelopers/colorado-rla/issues/73
+            // For now, tolerate > 1; later, check.
+            String winner = cr.getWinners().stream().findAny().orElse("UNKNOWN");
+            List<String> candidates = cr.getContests().stream().findAny().orElseThrow().choices().stream()
+                    .map(Choice::name).toList();
+
+            // Remove non-word characters for saving into .zip file; set up the zip next entry.
+            String sanitizedContestName = cr.getContestName().replaceAll("[\\W]", "");
+            zos.putNextEntry(new ZipEntry(sanitizedContestName + "_assertions." + suffix));
+
+            // Make the request.
+            GetAssertionsRequest getAssertionsRequest = new GetAssertionsRequest(
+                    cr.getContestName(),
+                    cr.getBallotCount().intValue(),
+                    candidates,
+                    winner,
+                    riskLimit
+            );
+            // TODO possibly we should be checking whether a valid URL has been set before we call it.
+            // Need to make sure that the exceptions we catch match the ones this will throw if that is wrong.
+            // See https://github.com/orgs/DemocracyDevelopers/projects/1/views/1?pane=issue&itemId=68081868
+            HttpPost requestToRaire = new HttpPost(raireUrl + "-" + suffix);
+            requestToRaire.addHeader("content-type", "application/json");
+            requestToRaire.setEntity(new StringEntity(gson.toJson(getAssertionsRequest)));
+
+            // Send it to the RAIRE service.
+            HttpResponse raireResponse = httpClient.execute(requestToRaire);
+            LOGGER.debug(String.format("%s %s.", prefix, "Sent Assertion Request to Raire service for "
+                    + getAssertionsRequest.contestName));
+
```

Comment:
Should the 200 be a constant set somewhere?

Reply:
yes - now fixed, with a builtin.